src/software/new_geom/segment.h

```diff
@@ -13,83 +10,76 @@ class Segment final
      *
      * @param o new start point of segment
      */
-    void setSegStart(Point o)
-    {
-        start = o;
-    }
+    void setStart(Point o);
 
     /**
      * Gets the start point of the segment
      *
      * @return start point of segment
      */
-    const Point& getSegStart() const
-    {
-        return start;
-    }
+    const Point& getStart() const;
 
     /**
      * Sets the end point of the segment to o
      *
      * @param o new end point of segment
      */
-    void setEnd(Point o)
-    {
-        end = o;
-    }
+    void setEnd(Point o);
 
     /**
      * Gets the end point of the segment
      *
      * @return end point of segment
      */
-    const Point& getEnd() const
-    {
-        return end;
-    }
+    const Point& getEnd() const;
+
+    // TODO function headers
+    double length();
+
+    double lengthSquared();
 
     /**
      * Creates a degenerate Segment at (0, 0)
      */
-    inline explicit Segment() {}
+    inline explicit Segment(){};
 
     /**
      * Creates a Segment that starts and ends at the given points
      */
-    inline explicit Segment(const Point& start, const Point& end) : start(start), end(end)
-    {
-    }
+    inline explicit Segment(const Point& start, const Point& end){};
 
     /**
      * Creates a Segment that is reversed
      */
     inline Segment reverse() const
     {
         return Segment(end, start);
-    }
+    };
 
     /**
      * Makes a Vector out of this Segment.
      */
     inline Vector toVector() const
     {
         return end - start;
-    }
+    };
 
     inline double length() const
     {
         return (end - start).length();
-    }
+    };
 
     inline double slope() const
     {
         return (end.y() - start.y()) / (end.x() - start.x());
-    }
+    };
```

Comment:
these should also go in the .cpp and strange `;` here

Reply:
Moved definitions to .cpp, removed `;`